packages/monitors/test/OptimisticOracleContractMonitor.js

```diff
@@ -0,0 +1,339 @@
+const winston = require("winston");
+const sinon = require("sinon");
+
+const { toWei, hexToUtf8, utf8ToHex } = web3.utils;
+
+const { OptimisticOracleContractMonitor } = require("../src/OptimisticOracleContractMonitor");
+const { interfaceName, MAX_UINT_VAL } = require("@uma/common");
+const { getTruffleContract } = require("@uma/core");
+
+const {
+  OptimisticOracleEventClient,
+  SpyTransport,
+  lastSpyLogIncludes,
+  lastSpyLogLevel
+} = require("@uma/financial-templates-lib");
+
+const CONTRACT_VERSION = "latest";
+
+const OptimisticOracle = getTruffleContract("OptimisticOracle", web3, CONTRACT_VERSION);
+const OptimisticRequesterTest = getTruffleContract("OptimisticRequesterTest", web3, CONTRACT_VERSION);
+const Finder = getTruffleContract("Finder", web3, CONTRACT_VERSION);
+const IdentifierWhitelist = getTruffleContract("IdentifierWhitelist", web3, CONTRACT_VERSION);
+const Token = getTruffleContract("ExpandedERC20", web3, CONTRACT_VERSION);
+const AddressWhitelist = getTruffleContract("AddressWhitelist", web3, CONTRACT_VERSION);
+const Timer = getTruffleContract("Timer", web3, CONTRACT_VERSION);
+const Store = getTruffleContract("Store", web3, CONTRACT_VERSION);
+const MockOracle = getTruffleContract("MockOracleAncillary", web3, CONTRACT_VERSION);
+
+contract("OptimisticOracleContractMonitor.js", function(accounts) {
+  const owner = accounts[0];
+  const requester = accounts[1];
+  const proposer = accounts[2];
+  const disputer = accounts[3];
+
+  let optimisticRequester;
+  let optimisticOracle;
+  let eventClient;
+  let contractMonitor;
+  let spyLogger;
+  let mockOracle;
+  let contractProps;
+  let monitorConfig;
+  let spy;
+
+  let finder;
+  let timer;
+  let identifierWhitelist;
+  let collateralWhitelist;
+  let store;
+  let collateral;
+
+  // Timestamps that we'll use throughout the test.
+  let requestTime;
+  let startTime;
+  let proposalTime;
+
+  // Default testing values.
+  const liveness = 7200; // 2 hours
+  const initialUserBalance = toWei("100");
+  const finalFee = toWei("1");
+  const reward = toWei("3");
+  // Proposal|Dispute bond = 2 x final fee = 2
+  const proposalPayout = toWei("5"); // dispute bond + reward
+  const disputePayout = toWei("5.5"); // dispute bond + 50% of loser's bond + reward
```

Comment:
Should this and `proposalPayout` be calculated based on `reward` and dispute bond instead of hardcoded with comments? Also, where is the dispute bond defined?

Reply:
Yep

Reply:
Also dispute bond = proposal bond, i'll make a note of that